tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -689,6 +689,42 @@ def test_log_softmax_invalid(device, a_shape, axis, dtype):
         return chainerx.log_softmax(a, axis=axis)
 
 
+@op_utils.op_test(['native:0', 'cuda:0'])
+@pytest.mark.parametrize('input', [
+    numpy.asarray(0), numpy.asarray(-1), numpy.asarray(1), numpy.asarray(
+        10), numpy.asarray(float('inf')), numpy.asarray(float('nan')),
+    numpy.full((), 2), numpy.full((0,), 2), numpy.full((2, 3), 2)
+])
+@pytest.mark.parametrize('contiguous', [None, 'C'])
+@pytest.mark.parametrize('in_dtype,out_dtype', _expected_dtypes_math_functions)
+class TestSquare(op_utils.NumpyOpTest):
+
+    def setup(self, input, contiguous, in_dtype, out_dtype):
+        self.input = input.astype(in_dtype)
+        self.chx_dtype = out_dtype
+        self.contiguous = contiguous
+
+        # Backward and double backward tests skipped to avoid
+        # numerical gradient calculation problem
+        if numpy.isinf(input).any():
+            op_utils.NumpyOpTest.skip_backward_test = True
+            op_utils.NumpyOpTest.skip_double_backward_test = True
```

Comment:
Oh, you should not update the class variable but the variable bound to this object, i.e.
```suggestion
            self.skip_double_backward_test = True
```
and ditto for the line above.

Reply:
Oh yes. Thanks for pointing that out @hvy .